added data and documentation for the womens-world-cup-2019 datasets

sunniraleigh opened the pull request:
added data and documentation for the two datasets (wwc_forecasts and wwc_matches) included in womens-world-cup-2019

sunniraleigh committed:
added data and documentation for the womens-world-cup-2019 datasets (wwc_matches and wwc_forecasts)

sunniraleigh commented on R/process_data_set_sunni.R:
```diff
@@ -0,0 +1,36 @@
+library(snakecase)
+
+## wwc_matches ---------------------------------------------------------------------------------------------------------
+wwc_2019_matches <- readr::read_csv("https://projects.fivethirtyeight.com/soccer-api/international/2019/wwc_matches.csv") %>%
```
Should the wwc_2019_matches data be tidier?

sunniraleigh commented on R/data_sunni.R:
```diff
@@ -0,0 +1,71 @@
+#'   \item{xg1}{Shot-based expected goals for team 1}
+#'   \item{xg2}{Shot-based expected goals for team 2}
+#'   \item{nsxg1}{Non-shot expected goals for team 1}
+#'   \item{nsxg2}{Non-shot expected goals for team 2}
```
Should the variables xg1, xg2, nsxg1, and nsxg2 have more descriptive definitions?

beanumber replied:
I think this is OK.

beanumber commented on R/process_data_set_sunni.R:
```diff
@@ -0,0 +1,36 @@
+    ) %>%
+  select(-c(forecast_timestamp, timestamp))
+
+wwc_2019_forecasts <- wwc_2019_forecasts[c(
```
@sunniraleigh What is this doing? Reordering the variables? Is there a reason you are using `[` instead of `select()` (as above)?

sunniraleigh replied:
I wanted to reorder the variables so that my new mutated variable was in the front.

beanumber replied:
OK. Perhaps @rudeboybert can weigh in on whether he wants the variable order to match what FiveThirtyEight has.

rudeboybert replied:
Hi @sunniraleigh, please try to keep the same order as the original 538 data if possible

beanumber commented on R/process_data_set_sunni.R:
```diff
@@ -0,0 +1,36 @@
+  "make_round_of_16", "make_quarters", "make_semis", "make_final", "win_league"
+)]
+
+usethis::use_data(wwc_2019_forecasts, overwrite = TRUE)
```
Why do you have four data files in `data/` below? I think you only want the two with `_2019_` in the title. If so, please delete the other two and the commit the deletion.

beanumber replied:
See also #54 and #56 for instructions about adding yourself to `DESCRIPTION` and amending `NEWS.md`.

rudeboybert replied:
Also, this file should be in the `data-raw/` folder, as all the data processing takes place there. Note that the contents of `data-raw/` are used to create all the `.rda` files in the `data/` folder, and not in the actual pkg build. Hence if you look at `.Rbuildignore`, `data-raw/` is included telling R not to consider this folder.

rudeboybert commented:
Thanks @sunniraleigh for your work. Also remember that you don't need to delete this current PR #59 and start a new one when making changes. Just commit/push your changes and this PR will update itself.

sunniraleigh committed:
deleted unnecessary data files, moved data process file, added myself to description and news

sunniraleigh committed:
readjusted columns

rudeboybert commented:
Hey @sunniraleigh, before I go over your changes, could you resolve the above merge conflict in the `DESCRIPTION` and `NEWS.md` files and commit/push them? I believe @beanumber has been suggesting students do a "git fetch upstream" type command, then fix all conflicts

beanumber commented:
Yes, @sunniraleigh , please try this:

https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/syncing-a-fork

sunniraleigh committed:
resolved merge conflict

Merge remote-tracking branch 'upstream/master'

# Conflicts:
#	DESCRIPTION
#	NEWS.md

rudeboybert commented on DESCRIPTION:
```diff
@@ -21,7 +21,8 @@ Authors@R: c(
     person("Jayla", "Nakayama", role = "ctb"),
     person("Natalia", "Iannucci", email = "iannuccinatalia@gmail.com", role = "ctb"),
-    person("Marium", "Tapal", email = "mariumtapal@gmail.com", role = "ctb"))
+    person("Marium", "Tapal", email = "mariumtapal@gmail.com", role = "ctb")),
```
Hey @sunniraleigh, the mismatched parentheses is causing the package build to fail. Please be sure to both:
* Build the package
* Run the package "check"
and ensure all errors are resolved before making any commits

rudeboybert replied:
@beanumber

sunniraleigh committed:
Fixed paratheses problem

rudeboybert commented on data-raw/womens-world-cup-2019/process_data_set_sunni.R:
```diff
@@ -0,0 +1,31 @@
+library(tidyverse)
```
Could you move this file one directory up? The `process_data_sets_X.R` files should be in `data-raw/` and not in the data subdirectories.

rudeboybert commented on data-raw/womens-world-cup-2019/process_data_set_sunni.R:
```diff
@@ -0,0 +1,31 @@
+  mutate(
+    team1 = to_snake_case(team1),
+    team2 = to_snake_case(team2)
+    )
```
Could you remove the `_women` from the `team1` and `team2` variables? You should be able to do this `stringr::str_replace_all()` within the `mutate()`

rudeboybert commented on data-raw/womens-world-cup-2019/process_data_set_sunni.R:
```diff
@@ -0,0 +1,31 @@
+    team = as.factor(team)
+    ) %>%
+  rename(date = forecast_timestamp) %>%
+  select(-timestamp)
```
Could you add an `arrange(date, group)` here so that the data is in chronological order, and then for each day sorted by `group`?

rudeboybert reviewed (commented):
Almost done! Once you make these last few changes, I'll merge your PR into `master`.

sunniraleigh committed:
moved file, mutated team1 and team2 variables, arranged forecasts data

rudeboybert committed:
Merge branch 'master' into master

rudeboybert commented:
Thanks @sunniraleigh! You're now listed as a contributor: https://github.com/rudeboybert/fivethirtyeight/graphs/contributors
